doc/examples/streamline_registration.py

```diff
@@ -208,43 +213,40 @@
 """
 
 from dipy.io.streamline import load_trk
-
 streamlines, hdr = load_trk('lr-superiorfrontal.trk')
 
 
 """
 We then apply the obtained deformation field to the streamlines. We first
-apply the non-rigid warping and then apply the computed affine transformation
-whose extents must be corrected to account for the different voxel grids of
-the moving and static images:
+apply the non-rigid warping and simultaneously apply a computed rigid affine
+transformation whose extents must be corrected to account for the different
+voxel grids of the moving and static images:
 
 """
+from dipy.tracking.streamline import deform_streamlines
 
-from dipy.tracking.streamline import deform_streamlines, Streamlines
-
-from dipy.tracking.utils import move_streamlines
-
-warped_streamlines = \
-    deform_streamlines(streamlines,
-                       deform_field=mapping.get_forward_field(),
-                       stream_to_current_grid=moving_affine,
-                       current_grid_to_world=mapping.codomain_grid2world,
-                       stream_to_ref_grid=static_affine,
-                       ref_grid_to_world=mapping.domain_grid2world)
+# Create an isocentered affine
+target_isocenter = np.diag(np.array([-vox_size, vox_size, vox_size, 1]))
 
+# Take the off-origin affine capturing the extent contrast between fa image and the template
+origin_affine = affine_map.affine.copy()
 
-adjusted_affine = moving_affine.copy()
-adjusted_affine[0][3] = -adjusted_affine[0][3] - static_affine[0][3]
-adjusted_affine[1][3] = -adjusted_affine[1][3] + static_affine[1][3]
-adjusted_affine[2][3] = -adjusted_affine[2][3] + static_affine[2][3]
+# Now we flip the sign in the x and y planes so that we get the mirror image of the forward deformation field.
+origin_affine[0][3] = -origin_affine[0][3]
+origin_affine[1][3] = -origin_affine[1][3]
 
-rotation, scale = np.linalg.qr(affine_map.affine)
-streams = move_streamlines(streamlines, rotation)
-scale[0:3, 0:3] = np.dot(scale[0:3, 0:3], np.diag(1. / hdr['voxel_sizes']))
-scale[0:3, 3] = abs(scale[0:3, 3])*1. / hdr['voxel_sizes'][0]
+# Scale z by the voxel size
+origin_affine[2][3] = origin_affine[2][3]/vox_size
 
-affine_streamlines = Streamlines(move_streamlines(warped_streamlines, scale))
+# Scale y by the square of the voxel size since we've already scaled along the z-plane.
+origin_affine[1][3] = origin_affine[1][3]/vox_size**vox_size
```

Comment:
I'll make the comments for the flips and scalings be rendered in the example. I think they are part of the things that are relevant to the reader.

@dPys May be adding additional rationale as to why they are required would be nice. To be honest, this is a part that I paid little attention when pushing the code, so myself has a hard time answering why they are the way they are (why do we need the mirror, why the y has to be scaled by the square, why x is not scaled, etc.) 😅 .

Reply:
Sounds good. The rationale is fairly complicated RE: scaling y and z dims but not x. Essentially what we are trying to do here is align the FOV of the template and the mirror image of the streamlines. To do this, we first need to flip the sign on the x-offset and y-offset. We need to use the information about the origin offsets (i.e. between the static and moving images) that we obtained using transform_origins. The trouble is that transform_origins returns this affine transformation with (1, 1, 1) zooms and not (2, 2, 2), which means that the offsets need to be scaled by 2. But when we scale the z-offset, we are also implicitly scaling the y-offset as well (by 1/2), thus we need to correct for this by only scaling the y by 1/4 and not 1/2.

There may be an easier solution to all of this, but I still haven't found it yet. One thing that might be worth trying down the road is simply feeding something like the inverse of the forward deformation field into deform_streamlines so that we don't need to worry about aligning the mirror-image FOV!

@dPys

Reply:
Uops, does not look evident. Thanks for the explanation Derek.